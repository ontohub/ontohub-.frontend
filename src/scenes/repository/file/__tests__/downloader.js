import FileSaver from "file-saver";
import { Downloader } from "../downloader";

jest.mock("file-saver");

describe("DownloadButton", () => {
  set("component", () => (
    <Downloader
      filename="test.js"
      encoding={encoding}
      value="This is the test content"
    />
  ));
  set("renderedComponent", () => render(component));
  set("mountedComponent", () => mount(component));

  describe("with a plain text file", () => {
    set("encoding", () => "plain");

    it("matches the snapshot", () => {
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe("with a base64-encoded text file", () => {
    set("encoding", () => "base64");

    it("matches the snapshot", () => {
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe("clicking the button", () => {
    beforeAll(() => {
      FileSaver.saveAs = jest.fn();
    });

    it("calls saveAs", () => {
      mountedComponent.find("Button").simulate("click");
      expect(FileSaver.saveAs.mock.calls.length).toBe(1);
    });
  });
});
