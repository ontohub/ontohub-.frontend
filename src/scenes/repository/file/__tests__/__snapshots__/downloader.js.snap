// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadButton with a base64-encoded text file matches the snapshot 1`] = `
<div
  class="ui one column grid"
  style="padding:1em"
>
  <div
    class="row"
  >
    <div
      class="center aligned column"
    >
      <div
        class="ui container"
      >
        This file type cannot be displayed.
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="center aligned column"
    >
      <button
        class="ui button"
        role="button"
      >
        Download test.js
      </button>
    </div>
  </div>
</div>
`;

exports[`DownloadButton with a plain text file matches the snapshot 1`] = `
<div
  class="ui one column grid"
  style="padding:1em"
>
  <div
    class="row"
  >
    <div
      class="center aligned column"
    >
      <div
        class="ui container"
      >
        This file type cannot be displayed.
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="center aligned column"
    >
      <button
        class="ui button"
        role="button"
      >
        Download test.js
      </button>
    </div>
  </div>
</div>
`;
